ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/DebugFaucet.java

```diff
@@ -0,0 +1,139 @@
+import org.hyperledger.besu.ethereum.core.Wei;
+import org.hyperledger.besu.ethereum.eth.transactions.TransactionPool;
+import org.hyperledger.besu.ethereum.mainnet.ValidationResult;
+import org.hyperledger.besu.ethereum.rlp.RLPException;
+import org.hyperledger.besu.ethereum.transaction.TransactionInvalidReason;
+
+import java.math.BigInteger;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import com.google.common.base.Suppliers;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+
+public class DebugFaucet implements JsonRpcMethod {
+  private static final com.google.common.base.Supplier<SignatureAlgorithm> SIGNATURE_ALGORITHM =
+      Suppliers.memoize(SignatureAlgorithmFactory::getInstance);
+  private static final long FAUCET_ACCESS_INTERVAL_SECONDS = TimeUnit.HOURS.toSeconds(6);
+
+  private final Supplier<BlockchainQueries> blockchainQueries;
+  private final Supplier<TransactionPool> transactionPool;
+  private final BigInteger chaindId;
+  private final KeyPair faucetKeyPair;
+  private final Address faucetAddress;
+  final Wei faucetAmount;
+  private final Map<String, Instant> access = new HashMap<>();
+
+  public DebugFaucet(
+      final BlockchainQueries blockchainQueries,
+      final TransactionPool transactionPool,
+      final BigInteger chaindId,
+      final String faucetPrivateKey,
+      final Wei faucetAmount) {
+    this.blockchainQueries = Suppliers.ofInstance(blockchainQueries);
+    this.transactionPool = Suppliers.ofInstance(transactionPool);
+    this.chaindId = chaindId;
+    this.faucetKeyPair = keyPair(faucetPrivateKey);
+    this.faucetAddress = Util.publicKeyToAddress(this.faucetKeyPair.getPublicKey());
+    this.faucetAmount = faucetAmount;
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.DEBUG_FAUCET.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
+    if (requestContext.getRequest().getParamLength() != 1) {
+      return new JsonRpcErrorResponse(
+          requestContext.getRequest().getId(), JsonRpcError.INVALID_PARAMS);
+    }
+    final String recipientAddress = requestContext.getRequiredParameter(0, String.class);
+
+    if (!allowed(recipientAddress)) {
+      return new JsonRpcErrorResponse(
+          requestContext.getRequest().getId(), JsonRpcError.UNAUTHORIZED);
+    }
+
+    final Transaction transaction;
+    try {
+      final long nonce = this.blockchainQueries.get().getTransactionCount(faucetAddress);
+      transaction =
+          Transaction.builder()
+              .chainId(chaindId)
+              .nonce(nonce)
+              .value(faucetAmount)
+              .gasLimit(30000)
+              .payload(Bytes.EMPTY.trimLeadingZeros())
+              .gasPrice(Wei.of(1000))
+              .to(Address.fromHexString(recipientAddress))
+              .guessType()
```

Comment:
The intention behind `guessType` was that it would be used when we're parsing transactions from json and they don't explicitly say the transaction type. I think we should be explicit whenever we're building the transaction.